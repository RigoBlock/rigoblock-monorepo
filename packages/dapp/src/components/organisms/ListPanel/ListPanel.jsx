import React from 'react'
import PropTypes from 'prop-types'
import ContentWrapper from '../../molecules/ContentWrapper'
import ListItem from '../../molecules/ListItem'

const ListPanel = props => {
  const { title, items, tooltip } = props
  const listItems = items.map(item => (
    <ListItem
      key={item.id}
      itemName={item.name}
      itemSymbol={item.symbol}
      itemValue={item.value}
      growth={item.growth}
      currencyGrowth={item.currencyGrowth}
    />
  ))
  return (
    <ContentWrapper title={title} tooltip={tooltip}>
      {listItems}
    </ContentWrapper>
  )
}

ListPanel.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  tooltip: PropTypes.string
}

ListPanel.defaultProps = {
  tooltip: null
}

export default ListPanel
