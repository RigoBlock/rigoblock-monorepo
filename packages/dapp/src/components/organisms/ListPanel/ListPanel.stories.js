import React from 'react'
import { storiesOf } from '@storybook/react'
import ListPanel from './ListPanel'
import '../../_settings/_base.scss'

const items = [
  {
    id: 1,
    name: 'Rocksolid Vault',
    symbol: 'VLT',
    value: 12489.51323
  },
  {
    id: 2,
    name: 'Big Drago',
    symbol: 'DRG',
    growth: 0.25,
    currencyGrowth: 2.32
  },
  {
    id: 3,
    name: 'Rocksolid Vault',
    symbol: 'VLT',
    value: 12489.51323
  }
]

storiesOf('Organisms/ListPanel', module)
  .addDecorator(story => <div style={{ width: '400px' }}>{story()}</div>)
  .add('default', () => <ListPanel title={'Example ListPanel'} items={items} />)
