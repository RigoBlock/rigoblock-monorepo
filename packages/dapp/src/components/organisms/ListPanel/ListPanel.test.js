import { shallow } from 'enzyme'
import ListPanel from './ListPanel.jsx'
import toJson from 'enzyme-to-json'

const props = {
  title: 'ListPanel Component',
  items: [
    {
      id: 1,
      name: 'Rocksolid Vault',
      symbol: 'VLT',
      value: 12489.51323
    }
  ]
}

describe('ListPanel component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(ListPanel, props)))
    ).toMatchSnapshot()
  })
})
