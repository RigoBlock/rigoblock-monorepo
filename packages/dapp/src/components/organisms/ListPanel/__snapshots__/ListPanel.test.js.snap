// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPanel component renders correctly 1`] = `
<ContentWrapper
  header={
    <PanelHeader
      title="ListPanel Component"
      tooltip={null}
    />
  }
>
  <List
    Component={[Function]}
    className={null}
    data={
      Array [
        Object {
          "id": 1,
          "itemName": "Rocksolid Vault",
          "itemSymbol": "VLT",
          "value": 12489.51323,
        },
      ]
    }
  />
</ContentWrapper>
`;
